source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -844,10 +842,15 @@ on a dedicated connection, for example:
             setUpConnection()
             return RTT of the connection handshake
 
+        if helloOk:
+            helloCommand = hello
+        else
+            helloCommand = legacy hello
+
         start = time()
-        # NOTE: helloCommand is defined in __init__() above
-        call {helloCommand: 1}
+        response = call {helloCommand: 1, helloOk: True}
         rtt = time() - start
+        helloOk = response.helloOk
```

Comment:
Valid point and great catch. I've reworked the `RttMonitor` logic to take `versionedApi` as a ctor param so that it can be properly reset to its original state when an exception occurs. This should let us handle a server downgrade correctly. Before the `RttMonitor` could be stuck using `hello` even if the downgraded server no longer supported it.